### PR TITLE
feat(core): Get and set username

### PR DESCRIPTION
**Pull request by ffflorian**

### CODE REVIEW

File: packages/core/src/main/self/SelfService.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  public async getName(): Promise<string> {
+    const {name} = await this.apiClient.self.api.getName();
+    return name;
+  }
+
+  public getSelf(): Promise<Self> {
+    return this.apiClient.self.api.getSelf();
+  }
+
+  public async setUserName(userName: string): Promise<{}> {
```
Comment (bennyn) [inline packages/core/src/main/self/SelfService.ts]: Let's also add `getUserName`

File: packages/core/src/main/self/SelfService.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    const [availableHandle] = await this.apiClient.user.api.postHandles({
+      handles: [userName],
+    });
+
+    if (availableHandle) {
+      return this.apiClient.self.api.putHandle({handle: userName});
+    }
+
+    throw new Error(`Username "${userName}" is not available.`);
+  }
```
Comment (gregor) [inline packages/core/src/main/self/SelfService.ts]: @ffflorian Why do you check availability when setting? `PUT` on the handle will also fail in this case. No need for the roundtrip before. The check of availability is used for UI purposes so we can show checkmarks and user feedback that the entered handle could be selected and set.

Comment (ffflorian) [reply packages/core/src/main/self/SelfService.ts]: Thanks! Will add a PR for the fix.

Comment (ffflorian) [reply packages/core/src/main/self/SelfService.ts]: https://github.com/wireapp/wire-web-packages/pull/568

**Pull request merged**